lib/features/in_app_review/business/in_app_rating_service.dart

```diff
@@ -0,0 +1,59 @@
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:in_app_review/in_app_review.dart";
+import "package:shared_preferences/shared_preferences.dart";
+
+import "../../../config/shared_prefs.dart";
+import "../../../utils/datetime_utils.dart";
+import "../../app_streak/business/get_days_use_case.dart";
+
+class InAppRatingService {
+  const InAppRatingService(this.ref);
+
+  final Ref ref;
+
+  InAppReview get _inAppReview => InAppReview.instance;
+
+  Future<int> _getUsageDays() async {
+    return await ref.read(getUsageDaysUseCaseProvider.future);
+  }
+
+  SharedPreferences get _sharedPrefs =>
+      ref.read(sharedPreferencesSingletonProvider).requireValue;
+
+  Future<void> requestReviewIfNeeded() async {
+    if (await _canRequestReview()) {
+      await _inAppReview.requestReview();
+      await _updatePreferences();
+    }
+  }
+
+  Future<bool> _canRequestReview() async {
+    final lastReviewPrompt = _sharedPrefs.getInt(
+      InAppReviewConfig.lastReviewPromptDateInMillis,
+    );
+    final reviewCount =
+        _sharedPrefs.getInt(InAppReviewConfig.reviewCountKey) ?? 0;
+    final usageDays = await _getUsageDays();
+
+    if (usageDays > 2 && reviewCount < 3 && await _inAppReview.isAvailable()) {
+      final DateTime lastReviewPromptDate = lastReviewPrompt != null
+          ? DateTime.fromMillisecondsSinceEpoch(lastReviewPrompt)
+          : DateTime.now().subtract(const Duration(days: 8));
```

Comment:
imo taking 8 days in the past to then pass the if condition feels hacky to me and will be harder to maintain if we'll change something here. Maybe cleaner would be stating the logic in more straightforward way. Example:
```dart
if (lastReviewPrompt == null) {
        return true;
      }
      final DateTime lastReviewPromptDate =
          DateTime.fromMillisecondsSinceEpoch(lastReviewPrompt);
      if (DateTime.now().difference(lastReviewPromptDate).inDays > 7) {
        return true;
      }
```

Reply:
Yeah, excellent suggestion. I was trying to keep it simple and eventually ended up even more jumbled  😂
Your next suggestion about making extensions os `SharedPref `object made that part of code even more readable.